# Изучение закономерностей определяющих успешность игр

<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-проекта" data-toc-modified-id="Описание-проекта-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание проекта</a></span></li><li><span><a href="#Просмотр-и-предобработка-данных" data-toc-modified-id="Просмотр-и-предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Просмотр и предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Переименование-столбцов" data-toc-modified-id="Переименование-столбцов-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Переименование столбцов</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Заполнение-пропусков" data-toc-modified-id="Заполнение-пропусков-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Заполнение пропусков</a></span></li><li><span><a href="#Подсчет-суммарных-продаж-во-всех-регионах" data-toc-modified-id="Подсчет-суммарных-продаж-во-всех-регионах-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Подсчет суммарных продаж во всех регионах</a></span></li></ul></li><li><span><a href="#Исследовательский-анализ-данных" data-toc-modified-id="Исследовательский-анализ-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследовательский анализ данных</a></span><ul class="toc-item"><li><span><a href="#Сколько-игр-выпускалось-в-разные-годы" data-toc-modified-id="Сколько-игр-выпускалось-в-разные-годы-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Сколько игр выпускалось в разные годы</a></span></li><li><span><a href="#Просмотр-динамики-продаж-по-платформам" data-toc-modified-id="Просмотр-динамики-продаж-по-платформам-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Просмотр динамики продаж по платформам</a></span></li><li><span><a href="#Просмотр-динамики-продаж-за-актуальный-период" data-toc-modified-id="Просмотр-динамики-продаж-за-актуальный-период-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Просмотр динамики продаж за актуальный период</a></span></li><li><span><a href="#Просмотр-динамики-продаж-за-актуальный-период-ТОП-10-платформ" data-toc-modified-id="Просмотр-динамики-продаж-за-актуальный-период-ТОП-10-платформ-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Просмотр динамики продаж за актуальный период ТОП-10 платформ</a></span></li><li><span><a href="#Влияние-на-продажи-отзывов-пользователей-и-критиков-внутри-одной-платформы" data-toc-modified-id="Влияние-на-продажи-отзывов-пользователей-и-критиков-внутри-одной-платформы-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Влияние на продажи отзывов пользователей и критиков внутри одной платформы</a></span></li><li><span><a href="#Оценка-распределения-игр-по-жанрам" data-toc-modified-id="Оценка-распределения-игр-по-жанрам-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Оценка распределения игр по жанрам</a></span></li></ul></li><li><span><a href="#Составление-портрета-пользователя-каждого-региона" data-toc-modified-id="Составление-портрета-пользователя-каждого-региона-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Составление портрета пользователя каждого региона</a></span><ul class="toc-item"><li><span><a href="#Оценка-долей-продаж" data-toc-modified-id="Оценка-долей-продаж-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Оценка долей продаж</a></span></li><li><span><a href="#Оценка-самых-популярных-жанров" data-toc-modified-id="Оценка-самых-популярных-жанров-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Оценка самых популярных жанров</a></span></li><li><span><a href="#Оценка-влияния-рейтинга-ESRB-на-продажи-в-отдельном-регионе" data-toc-modified-id="Оценка-влияния-рейтинга-ESRB-на-продажи-в-отдельном-регионе-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Оценка влияния рейтинга ESRB на продажи в отдельном регионе</a></span></li></ul></li><li><span><a href="#Проверка-гипотез" data-toc-modified-id="Проверка-гипотез-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Проверка гипотез</a></span><ul class="toc-item"><li><span><a href="#Проверка-гипотезы:-средние-пользовательские-рейтинги-платформ-Xbox-One-и-PC-одинаковые" data-toc-modified-id="Проверка-гипотезы:-средние-пользовательские-рейтинги-платформ-Xbox-One-и-PC-одинаковые-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Проверка гипотезы: средние пользовательские рейтинги платформ Xbox One и PC одинаковые</a></span></li><li><span><a href="#Проверка-гипотезы:-средние-пользовательские-рейтинги-жанров-Action-и-Sports-разные." data-toc-modified-id="Проверка-гипотезы:-средние-пользовательские-рейтинги-жанров-Action-и-Sports-разные.-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Проверка гипотезы: средние пользовательские рейтинги жанров Action и Sports разные.</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

## Описание проекта

Вы работаете в интернет-магазине игр для платформ, который продаёт по всему миру компьютерные игры. Из открытых источников доступны исторические данные о продажах игр, оценки пользователей и экспертов, жанры и платформы (например, Xbox или PlayStation). Вам нужно выявить определяющие успешность игры закономерности. Это позволит сделать ставку на потенциально популярный продукт и спланировать рекламные кампании.
Перед нами данные до 2016 года. Сейчас декабрь 2016 г., и мы планируете кампанию на 2017-й интернет-магазина «Стримчик».


Описание данных
- Name — название игры
- Platform — платформа
- Year_of_Release — год выпуска
- Genre — жанр игры
- NA_sales — продажи в Северной Америке (миллионы проданных копий)
- EU_sales — продажи в Европе (миллионы проданных копий)
- JP_sales — продажи в Японии (миллионы проданных копий)
- Other_sales — продажи в других странах (миллионы проданных копий)
- Critic_Score — оценка критиков (максимум 100)
- User_Score — оценка пользователей (максимум 10)
- Rating — рейтинг от организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

Данные за 2016 год могут быть неполными.


Категории рейтинга, согласно ESRB:

- «EC» («Early childhood») — «Для детей младшего возраста»;
- «E» («Everyone») — «Для всех»;
- «E10+» («Everyone 10 and older») — «Для всех от 10 лет и старше»;
- «T» («Teen») — «Подросткам»;
- «M» («Mature») — «Для взрослых»;
- «AO» («Adults Only 18+») — «Только для взрослых».

## Просмотр и предобработка данных

Загрузка всех необходимых библиотек и задаем настройки

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats as st
import seaborn as sns
import warnings 
warnings.filterwarnings('ignore')
%config InlineBackend.figure_format = 'retina'  # задаем четкие цифры на графиках
from pylab import rcParams
rcParams['figure.figsize'] = 14,6 # задаем размер графиков, если не задано будет другого

Прочтем данные и запишем в переменную

In [ ]:
df_games = pd.read_csv('/datasets/games.csv')

Посмотрим информацию о данных

In [ ]:
df_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
Name               16713 non-null object
Platform           16715 non-null object
Year_of_Release    16446 non-null float64
Genre              16713 non-null object
NA_sales           16715 non-null float64
EU_sales           16715 non-null float64
JP_sales           16715 non-null float64
Other_sales        16715 non-null float64
Critic_Score       8137 non-null float64
User_Score         10014 non-null object
Rating             9949 non-null object
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


In [ ]:
df_games.describe()

Year_of_Release      NA_sales      EU_sales      JP_sales  \
count     16446.000000  16715.000000  16715.000000  16715.000000   
mean       2006.484616      0.263377      0.145060      0.077617   
std           5.877050      0.813604      0.503339      0.308853   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2016.000000     41.360000     28.960000     10.220000   

        Other_sales  Critic_Score  
count  16715.000000   8137.000000  
mean       0.047342     68.967679  
std        0.186731     13.938165  
min        0.000000     13.000000  
25%        0.000000     60.000000  
50%        0.010000     71.000000  
75%        0.030000     79.000000  
max       10.570000     98.000000

In [ ]:
df_games.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

   NA_sales  EU_sales  JP_sales  Other_sales  Critic_Score User_Score Rating  
0     41.36     28.96      3.77         8.45          76.0          8      E  
1     29.08      3.58      6.81         0.77           NaN        NaN    NaN  
2     15.68     12.76      3.79         3.29          82.0        8.3      E  
3     15.61     10.93      3.28         2.95          80.0          8      E  
4     11.27      8.89     10.22         1.00           NaN        NaN    NaN  
5     23.20      2.26      4.22         0.58           NaN        NaN    NaN  
6     11.28      9.14      6.50         2.88          89.0        8.5      E  
7     13.96      9.18      2.93         2.84          58.0        6.6      E  
8     14.44      6.94      4.70         2.24          87.0        8.4      E  
9     26.93      0.63      0.28         0.47           NaN        NaN    NaN

Вывод, необходимо:
- замените названия столбцов (привести к нижнему регистру);
- пропуски в столбцах critic_score, user_score и rating;
- привести данные к нужным типам year_of_release к целочисленному, user_score к числовому.

### Переименование столбцов

Приведем все столбцы к нижнему регистру

In [ ]:
df_games.columns = df_games.columns.str.lower()

In [ ]:
df_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
name               16713 non-null object
platform           16715 non-null object
year_of_release    16446 non-null float64
genre              16713 non-null object
na_sales           16715 non-null float64
eu_sales           16715 non-null float64
jp_sales           16715 non-null float64
other_sales        16715 non-null float64
critic_score       8137 non-null float64
user_score         10014 non-null object
rating             9949 non-null object
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


### Изменение типов данных

Посмотрим отдельно столбцы.
user_score - 'object', странно, если бы там были только числа и NaN, то формат был бы 'float'. Посмотрим все уникальные значения этого столбца

In [ ]:
df_games['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

Видим значение 'tbd'. Посмотрим сколько таких значений

In [ ]:
df_games[df_games['user_score'] == 'tbd']['user_score'].count()

2424

In [ ]:
df_games[df_games['user_score'] == 'tbd']['user_score'].count()/len(df_games['user_score'])*100

14.501944361352079

2424 строки с таким значением (14.5 процентов от всех значений)

Почитав игровые форму, выяснилось, что это аббревиатура "To Be Determined". Дается тогда, когда игра имеет недостаточное количество отзывов для подсчета рейтинга.
Такое может быть когда игры малопопулярны. 'tbd' фактически является NaN, заменим на NaN и приведем значение столбца user_score к числовому c плавающей точкой

In [ ]:
df_games.loc[df_games['user_score'] == 'tbd','user_score'] = np.nan
df_games['user_score'] = df_games['user_score'].astype('float')

Изменим тип данных в столбце year_of_release 

In [ ]:
df_games['year_of_release'] = df_games['year_of_release'].astype('Int16')

In [ ]:
df_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
name               16713 non-null object
platform           16715 non-null object
year_of_release    16446 non-null Int16
genre              16713 non-null object
na_sales           16715 non-null float64
eu_sales           16715 non-null float64
jp_sales           16715 non-null float64
other_sales        16715 non-null float64
critic_score       8137 non-null float64
user_score         7590 non-null float64
rating             9949 non-null object
dtypes: Int16(1), float64(6), object(4)
memory usage: 1.3+ MB


### Заполнение пропусков

Посмотрим сколько пропусков в каждом столбце

In [ ]:
df_games.isna().sum()

name                  2
platform              0
year_of_release     269
genre                 2
na_sales              0
eu_sales              0
jp_sales              0
other_sales           0
critic_score       8578
user_score         9125
rating             6766
dtype: int64

- critic_score - оценка критиков - вполне возможно, что игры не оценивались критиками. Было бы неправильно заполнить рейтинг игры средним или медианой, даже если бы мы их считали внутри для каждого жанра отдельно. Для дальнейшей обработки и оценки, поставим этим играм рейтинг -1 (вдруг у какой-то игры есть реально нулевой рейтинг, поставив -1, мы никогда не перемешаем игры с NaN и с реально нулевым рейтингом);
- rating - рейтинг от ESRB - невозможно дать рейтинг игре, посчитав его как среднее рейтингов игр того же жанра, к тому же категориальная переменная. Поставим рейтинг "No ESRB" - без рейтинга;
- user_score - или было tbd или просто игроки не ставили оценку. Поэтому и присутствуют пропуски. Поступим так же, как и со столбцом critic_score.

In [ ]:
# df_games['critic_score'] = df_games['critic_score'].fillna(-1)
# df_games['user_score'] = df_games['user_score'].fillna(-1)

In [ ]:
df_games.loc[df_games['user_score'] == 'tbd', 'user_score'] = float('nan')

In [ ]:
df_games['rating'].unique()

array(['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'], dtype=object)

In [ ]:
df_games['rating'] = df_games['rating'].fillna(0)

In [ ]:
df_games['rating'] = df_games['rating'].replace(0, 'No ESRB')

In [ ]:
df_games['rating'].unique()

array(['E', 'No ESRB', 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'],
      dtype=object)

In [ ]:
df_games.isna().sum()

name                  2
platform              0
year_of_release     269
genre                 2
na_sales              0
eu_sales              0
jp_sales              0
other_sales           0
critic_score       8578
user_score         9125
rating                0
dtype: int64

Посмотрим пропуски в годах выпуска

In [ ]:
df_games['year_of_release'].isna().sum()

269

Посмотрим наименования игр где пропущены года

In [ ]:
df_games[df_games['year_of_release'].isna() == True]['name']

183                        Madden NFL 2004
377                       FIFA Soccer 2004
456             LEGO Batman: The Videogame
475             wwe Smackdown vs. Raw 2006
609                         Space Invaders
                       ...                
16373    PDC World Championship Darts 2008
16405                        Freaky Flyers
16448                            Inversion
16458          Hakuouki: Shinsengumi Kitan
16522                         Virtua Quest
Name: name, Length: 269, dtype: object

Посмотрим есть ли на других платформах год выпуска

In [ ]:
df_games.query('name == "Madden NFL 2004"')

name platform  year_of_release   genre  na_sales  eu_sales  \
183   Madden NFL 2004      PS2              NaN  Sports      4.26      0.26   
1881  Madden NFL 2004       XB             2003  Sports      1.02      0.02   
3889  Madden NFL 2004       GC             2003  Sports      0.40      0.10   
5708  Madden NFL 2004      GBA             2003  Sports      0.22      0.08   

      jp_sales  other_sales  critic_score  user_score rating  
183       0.01         0.71          94.0         8.5      E  
1881      0.00         0.05          92.0         8.3      E  
3889      0.00         0.01          94.0         7.7      E  
5708      0.00         0.01          70.0         6.6      E

Заполним пробелы в годах за счет наличия информации об одних и тех же играх, только выпущенных для разных платформ

In [ ]:
for year in df_games[df_games['year_of_release'].isnull() == True].index:  
    df_games['year_of_release'][year] = df_games.loc[df_games['name'] == df_games['name'][year], 'year_of_release'].max()

In [ ]:
df_games['year_of_release'].isna().sum()

146

Часть пропусков в годах выпуска заполнили

In [ ]:
df_games = df_games.dropna(subset = ['year_of_release'])

In [ ]:
df_games['year_of_release'].isna().sum()

0

### Подсчет суммарных продаж во всех регионах

In [ ]:
df_games['global_sales'] = df_games['na_sales'] + df_games['eu_sales'] + df_games['jp_sales'] + df_games['other_sales']

Посмотрим таблицу

In [ ]:
df_games.head(10)

name platform  year_of_release         genre  \
0                 Wii Sports      Wii             2006        Sports   
1          Super Mario Bros.      NES             1985      Platform   
2             Mario Kart Wii      Wii             2008        Racing   
3          Wii Sports Resort      Wii             2009        Sports   
4   Pokemon Red/Pokemon Blue       GB             1996  Role-Playing   
5                     Tetris       GB             1989        Puzzle   
6      New Super Mario Bros.       DS             2006      Platform   
7                   Wii Play      Wii             2006          Misc   
8  New Super Mario Bros. Wii      Wii             2009      Platform   
9                  Duck Hunt      NES             1984       Shooter   

   na_sales  eu_sales  jp_sales  other_sales  critic_score  user_score  \
0     41.36     28.96      3.77         8.45          76.0         8.0   
1     29.08      3.58      6.81         0.77           NaN         NaN   
2     15.68     12.76      3.79         3.29          82.0         8.3   
3     15.61     10.93      3.28         2.95          80.0         8.0   
4     11.27      8.89     10.22         1.00           NaN         NaN   
5     23.20      2.26      4.22         0.58           NaN         NaN   
6     11.28      9.14      6.50         2.88          89.0         8.5   
7     13.96      9.18      2.93         2.84          58.0         6.6   
8     14.44      6.94      4.70         2.24          87.0         8.4   
9     26.93      0.63      0.28         0.47           NaN         NaN   

    rating  global_sales  
0        E         82.54  
1  No ESRB         40.24  
2        E         35.52  
3        E         32.77  
4  No ESRB         31.38  
5  No ESRB         30.26  
6        E         29.80  
7        E         28.91  
8        E         28.32  
9  No ESRB         28.31

Проверим на дубликаты на всякий случай

In [ ]:
df_games.duplicated().sum()

0

## Исследовательский анализ данных

### Сколько игр выпускалось в разные годы

Напишем функцию для построения гистограмм

In [1]:
def create_bar(groupby_column, func, y='name'):
    plt.style.use('seaborn-muted')
    df_to_plot = df_games.groupby(groupby_column)[y]
    if func == 'count':
        plot = df_to_plot.count().plot(kind='bar', y=y, ec='black')
    elif func == 'sum':
        plot = df_to_plot.sum().sort_values().plot(kind='barh', y=y, ec='black')

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Круто, что ты используешь функции для оптимизации кода
    
</div>

In [2]:
create_bar('year_of_release', 'count')
plt.xlabel('Год релиза', fontsize = 12)
plt.ylabel('Количество игр по годам', fontsize = 12);

NameError: ignored

Резкий скачок по количеству игр, выпущенных в год, произошел в 2002 году. Пик - 2008-2009 года. После этого начался спад, связанный с развитием мобильных устройств, интеграция соц. сетей в жизнь людей. Это всё привело к тому, что интерес к самим играм упал, соответственно, упало и их производство. Данные с 1980-1993 не особо значимы

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Наглядный график, понятные выводы
    
</div>

### Просмотр динамики продаж по платформам

In [ ]:
create_bar('platform', 'sum', 'global_sales')
plt.xlabel('Количество продаж', fontsize = 12)
plt.ylabel('Платформа', fontsize = 12);

Лидер - платформа PS2. Далее идут Xbox360, PS3 и Wii.
Самые редкие платформы - представители японского рынка

Выберем платформы с наибольшими суммарными продажами и построим распределение по годам. Возьмем топ 10 платформ

In [ ]:
top_platforms = df_games.groupby('platform').agg({'global_sales':'sum'}).sort_values(by='global_sales', ascending=False).head(10).index
for platform in top_platforms:
    df_games[df_games['platform'] == platform].pivot_table(index='year_of_release', values='global_sales', aggfunc='sum').plot(
        kind='bar')
    plt.title(platform)
    plt.style.use('seaborn-muted')
    plt.xlabel('Год релиза', fontsize = 12)
    plt.ylabel('Количество продаж в мире', fontsize = 12)
    plt.show()

Построим на одном графике

In [ ]:
df_games.query('platform in @top_platforms').pivot_table(index='year_of_release', columns = 'platform', 
                                                   values='global_sales', aggfunc='sum').plot(grid = True,style = 'o-', figsize=(15,6))
plt.xlabel('Год релиза', fontsize = 12);

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Хорошее решение построить на одном графике, так нагляднее
    
</div>

- средний период 'жизни' платформ длится 8-10 лет;
- лидирующих по продажам за всё время платформ в нашем датасете заметно идет на убыль к 2016 года (а то и раньше как общий лидер PS2 к 2011 году);
- к 2016 году остается на плаву только PS3 и то продаж очень мало;
- самая "прямая" зависимость у PC. Нет явных спадов и падений;
- у PS4 реже всего выходят релизы, зато в 2016 году она актуальна;
- такие платформы как PS2, PS, GDA уже до 2010 года перестали выпускать игры.

### Просмотр динамики продаж за актуальный период

Возьмем данные за соответствующий актуальный период. Основным фактором выбора актуального периода является то, что эти данные помогут построить прогноз на 2017 год.

Выделяются 3 этапа в индустрии: резкий рост с 2002 до 2008 года, пик 2008 и 2009 годов,и дальнейший упадок
Также большей части платформ уже нет в 2016 году, и соответственно в построении прогноза на 2017 год они нам не сильно помогут.
При уменьшении периода в рассмотрение попадут только последние поколения платформ.
Актуальный период возьмем с 2013 года, так падения продаж не ожидается

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

Склонен не согласиться с тобой относительно определения актуального периода.

Ты верно сказала, что стоит делать акцент на актуальных платформах, и поэтому стоит ограничиться максимум 2012 годом - годом до выхода последнего (на момент выгрузки данных) поколения консолей. Плюсом к этому в 2008 у нас пик по продажам, а далее до 2011 года идет спад, который все-таки стабилизировался.

Если считать от 2008 года, то стоит ожидать падения продаж в 2017 году, а если брать от 2013 года, то падение продаж не ожидается
    
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Спасибо, актуальный период возьмем с 2013 года </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Успех**

Спасибо, что прислушиваешься :)
    
</div>

In [ ]:
actual_period_data = df_games[df_games['year_of_release'] >= 2013]
actual_period_data

Построим ещё один график, отображающий динамику продаж по уже актуальным годам и ответим на вопрос: Какие платформы лидируют по продажам, растут или падают?

In [ ]:
actual_period_data.query('platform in @top_platforms').pivot_table(index='year_of_release', columns = 'platform', 
                                                   values='global_sales', aggfunc='sum').plot(grid = True, style = 'o-')
actual_period_data.query('platform in @top_platforms').pivot_table(index='year_of_release', columns = 'platform', 
                                                   values='global_sales', aggfunc='sum')

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

Ты потеряла XOne - очень важная платформа в актуальном периоде. Просто она не попала в топ-10 за все времена)
    
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Попробую не согласиться, ведь мы берем топ-10 не просто так, а по продажам за весь период, а XOne в него не попала, я как человек далекий от игровой индустрии не могу знать важна она или нет. для меня это платформа аж на 18 месте по продажам </div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Но ты совершенно прав что XOne на актуальный период имеет хорошие перспективы, просто она появилась позже остальных и поэтому за все время существования имеет не такие продажи как остальные, поэтому проведем анализ наоборот не как указано в задании к проекту, сначала возьмем актуальный период и уже в нем возьмем ТОП-10 (п.3.4).  </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Успех**

Да, стоит немного поменять формулировку проекта

Спасибо, что прислушалась
    
</div>

Посмотрим "срок жизни" платформ

In [ ]:
time_platforms = df_games.groupby('platform').agg({'year_of_release' : 'max'}) -  df_games.groupby('platform').agg({'year_of_release' : 'min'}) 
time_platforms

Видим, что XOne всего 3 года на рынке (на данные 2016 года)

<div class="alert alert-info"> <b>Комментарий студента:</b> Проведу анализ ТОП-10 за актуальный период </div>

### Просмотр динамики продаж за актуальный период ТОП-10 платформ

Выберем платформы с наибольшими суммарными продажами за актуальный перио 2013-2016 года и построим распределение по годам. Возьмем топ 10 платформ

In [ ]:
top_platforms_actual = actual_period_data.groupby('platform').agg({'global_sales':'sum'}).sort_values(by='global_sales', ascending=False).head(10).index
for platform in top_platforms_actual:
    actual_period_data[actual_period_data['platform'] == platform].pivot_table(index='year_of_release', values='global_sales', aggfunc='sum').plot(
        kind='bar')
    plt.title(platform)
    plt.style.use('seaborn-muted')
    plt.xlabel('Год релиза', fontsize = 12)
    plt.ylabel('Количество продаж в мире', fontsize = 12)
    plt.show()

In [ ]:
actual_period_data.query('platform in @top_platforms_actual').pivot_table(index='year_of_release', columns = 'platform', 
                                                   values='global_sales', aggfunc='sum').plot(grid = True, style = 'o-')
actual_period_data.query('platform in @top_platforms_actual').pivot_table(index='year_of_release', columns = 'platform', 
                                                   values='global_sales', aggfunc='sum')

На конец 2016 года лидируют по продажам следующие платформы:

- PS4;
- XOne
- 3DS;

Это топ-3. Причем количество продаж игр, выпущенных на каждую из этих платформ, снижается

Построим график «ящик с усами» по глобальным продажам игр в разбивке по платформам. 

In [ ]:
plt.figure(figsize=(12,10))
sns.boxplot(x = 'platform', y = 'global_sales', 
            data = actual_period_data[actual_period_data.platform.isin(top_platforms_actual)]);
plt.ylim(0,1.75)
plt.xlabel('Платформа', fontsize=12)
plt.ylabel('Глобальные продажи',fontsize=12);

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

Не стоит ограничивать график по высоте - так мы потеряем важные выбросы
    
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Но ведь если не ограничить по высоте графики станут просто нечитабельны </div>

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Совет**

Тогда можно посмотреть на 95 и 99 квантиль, и если 99 квантиль все-таки где-то в космосе, то можно его обрезать, и остальные данные все-таки вывести на графики

Квантили можно смотреть при помощи `df.column.quantile(n), где 0 < n < 1.`
    
</div>

- графиков видно, что медиана протекает у всех по разному. От 0.1 до 0.2 млн.;
- верхняя граница, до аномалий, PS4 - на уровне 1,55 млн - максимальные, дальше идет XOne фактически лидер последних (до 2016 года лет)-1,5 млн;
- у платформы 3DS,PC3, РС показатели ниже;
- у платформ PSV, PSP показатели еще ниже но на 2016 год они все еще актуальны

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

По выбросам тоже следует сделать выводы, вдруг какие-то платформы живут за счет бестселлеров
    
</div>

### Влияние на продажи отзывов пользователей и критиков внутри одной платформы

За образец возьмем PS3 - она давно на рынке, по ней много данных, при этом на 2016 год она все еще актуальна (хоть PS4 имеет большую популярность в 2016 году, но она на рынке относительно недавно)

Построим диаграмму рассеяния продаж от оценки пользователей и найдем их корреляцию

In [ ]:
plt.figure(figsize=(12,6))
sp3_user = df_games[(df_games['platform'] == 'PS3') & (df_games['user_score'] > 0)]
sp3_user.plot(kind = 'scatter', x = 'user_score', y = 'global_sales', alpha = 0.5)
plt.xlabel('Количество продаж', fontsize=12)
plt.ylabel('Оценка пользователей',fontsize=12)
corr_user = sp3_user['user_score'].corr(sp3_user['global_sales'])
print(f'Корреляция между продажами и оценкой пользователей: {corr_user}');

In [ ]:
plt.figure(figsize=(12,6))
sp3_critic = df_games[(df_games['platform'] == 'PS3') & (df_games['critic_score'] > 0)]
sp3_critic.plot(kind = 'scatter', x = 'critic_score', y = 'global_sales', alpha = 0.5)
plt.xlabel('Количество продаж', fontsize=12)
plt.ylabel('Оценка китиков',fontsize=12)
corr_critic = sp3_critic['critic_score'].corr(sp3_critic['global_sales'])
print(f'Корреляция между продажами и оценкой критиков:{corr_critic}');

Наблюдается не сильная связь между оценками критиков и продажами. Связь между оценками пользователей и продажами слабее.  Это говорит о том, что покупатели прислушиваются больше к критикам, нежели к оценкам других пользователей.

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Частично верно.

Также критики по большей части оценивают игру до её выхода. Чем больше критиков оценило игру высоко, тем скорее всего больше маркетинговый бюджет издателя :D

А отзывы игроков приходят уже тогда, когда игра уже продалась большим тиражом
    
</div>

Проверим данную зависимость для других платформ
Напишем функцию построения диаграммы рассеяния и подсчета корреляции

In [ ]:
def platform_corr_matrix(name_of_platform):
    platform = df_games[df_games['platform']==name_of_platform]
    fig, ax = plt.subplots(1 ,2, figsize=(15,5))
    sns.scatterplot(x='user_score', y='global_sales', data=platform, ax=ax[0])
    sns.scatterplot(x='critic_score', y='global_sales', data=platform, ax=ax[1])
    fig.suptitle(name_of_platform, fontsize=15)
    ax[0].set(xlabel='Оценка пользователей')
    ax[1].set(xlabel='Оценка критиков')
    ax[0].set(ylabel='Количество продаж')
    ax[1].set(ylabel='Количество продаж')
    plt.show()
    
    correl = platform['user_score'].corr(platform['global_sales'])
    critic_correl = platform['critic_score'].corr(platform['global_sales'])
    print(f'Корреляция между продажами и оценкой пользователей:{correl}')
    print(f'Корреляция между продажами и оценкой критиков:{critic_correl}')

In [ ]:
for platform in top_platforms:
    platform_corr_matrix(platform)

Для платформ PS2, X360 зависимость похожа на зависимость PS3: корреляция между оценкой критиков и продажами сильнее чем корреляция между оценками пользователей и продажами. У платформ Wii, PS4 также корреляция между оценкой критиков и продажами сильнее чем корреляция между оценками пользователей и продажами, но сама корреляция слабее. У платформ PC, GBA корреляция очень слабая. И только PSP у корреляция между оценками пользователей и продажами сильнее чем корреляция между оценкой критиков (у DS зависимость такая же но слабая). 

В целом связь между оценками пользователей и продажами и оценками критиков и продажами слабая у всех платформ, либо её совсем нет. 

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Все верно
    
</div>

### Оценка распределения игр по жанрам

In [ ]:
ax = plt.gca()

pivot = df_games.groupby('genre').agg({'name': 'count', 'global_sales': 'sum'}).sort_values(by='name', ascending=False)
plot1 = pivot['name']
plot1.plot(kind='bar', figsize=(15,5), ec='black', ax=ax, width=0.2, position=1)
#plot2 = pivot['global_sales']
#plot2.plot(kind='bar', figsize=(15,5), ec='black', ax=ax, width=0.2, color='lightgreen', position=0)
#ax.legend(['Количество продаж', 'Общая сумма продаж'])
plt.xlabel('Жанр', fontsize=12);

In [ ]:
most_popular_genre = df_games.groupby('genre').agg({'name':'count'}).reset_index()
most_popular_genre.sort_values(by='name', ascending=False).head()

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

В этом графике легко запутаться. 

Что такое количество продаж? Я это понимаю, но поймет ли заказчик отчета. 

Плюс у тебя один показатель считается в штуках (уникальных играх), а второй - в миллионах копий. 
    
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Убрала количество продаж </div>

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

Как вариант - можно было оставить оба, но сделать в легенде подписи, в каких величинах смотрим
    
</div>

И по количеству и по сумме продаж первое место занимают игры жанра action, меньше всего продаж у жанра puzzle, а меньше всего сумма продаж у жанра adventure (видимо цена игр не велика)

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

У нас в данных никакой цены игр нет. Есть миллионы проданных копий.
    
</div>

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

Стоит добавить средние и медианные продажи по платформам за актуальный период. Вдруг жанр живет за счет бестселлеров, а без них остальной жанр мертв. 

Частично это можно понять по твоему графику выше, но стоит все-таки сделать на этом акцент и вывести таблицей
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Добавила </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Успех**

Отлично
    
</div>

In [ ]:
pivot_actual = actual_period_data.pivot_table(index='genre', 
                               values='global_sales',aggfunc=['mean','median'])
pivot_actual.columns = ['mean', 'median']
pivot_actual.sort_values(by='mean',ascending=False)

Среднее и медианное значение больше всего у жанра shooter

Посмотрим динамику продаж по годам в зависимости от жанра

In [ ]:
actual_period_data.pivot_table(index='year_of_release', columns = 'genre', values='global_sales', 
                               aggfunc='count').plot(grid = True, style = 'o-')
actual_period_data.pivot_table(index='year_of_release', columns = 'genre', values='global_sales', 
                               aggfunc='count').sort_values(by='year_of_release', ascending=False)
plt.style.use('ggplot')

В целом все жанры падают в продажах из года в год. Чуть растет жанр racing и shooter.

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

У тебя узкая цветовая палитра, не совсем понятно, где какой жанр)
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Подскажи пожалуйста как задать палитру. и можешь, пожалуйста, еще подсказать как сделать так чтобы не надо было прокручивать окно вывода графиков </div>

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Совет**

Должна помочь вот такая настройка `plt.style.use('ggplot')`
У ggplot пошире цветовая палитра
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Спасибо </div>

## Составление портрета пользователя каждого региона

### Оценка долей продаж

Напишем функцию для построения графиков в зависимости от того что группируем

In [ ]:
def top_in_regions(groupby, region_sales, ax):
    pivot = actual_period_data.groupby(groupby).agg({region_sales: 'sum'}).sort_values(by=region_sales, ascending=False).head(5)
    title = {'na_sales': 'Продажи в Северной Америке', 'eu_sales': 'Продажи в Европе', 'jp_sales': 'Продажи в Японии'}
    color = {'na_sales': 'blue', 'eu_sales': 'red', 'jp_sales': 'green'}
    plot = pivot.plot(kind='bar', ec='black', title=title[region_sales], ax=axes[ax], 
                   fontsize=18, color=color[region_sales], rot=30)
    plot.set_ylabel('Млн. копий', fontsize = 15)
    plot.set_xlabel('')
    plot.title.set_size(20)

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

Тут ты смотришь на весь период, а не только актуальный.
    
</div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Функция просто супер
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Поменяла период. Спасибо)</div>

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

Посоветую заменить столбчатые графики круговой диаграммой, при этом не надо исключать все остальные платформы за пределами топ-5. 

Вдруг топ-5 платформ занимают только 20% рынка конкретного региона, а остальные 80% - куча консолей с маленькими тиражами
    
</div>

In [ ]:
fig, axes = plt.subplots(1, 3, figsize=(25,6))
top_in_regions('platform', 'na_sales', 0)
top_in_regions('platform', 'eu_sales', 1)
top_in_regions('platform', 'jp_sales', 2)

В Северной Америке популярнее всего PS4 и XOne, самая не популярная (из топ 5) это PS3.
В Европе самая популярная, как и в Америке, это PS4 и XOne(но отствет значительнее от первого места, чем в Америке) самая не популярная - 3DS.
В Японии противоположная ситуация: самые непопулярные платформы у Европы и Северной Америке, тут на первом месте - 3DS, а PS4 только на 2, популярный же в Америке XOne даже в топ-5 не зашел.

### Оценка самых популярных жанров

In [ ]:
fig, axes = plt.subplots(1, 3, figsize=(25,6))
top_in_regions('genre', 'na_sales', 0)
top_in_regions('genre', 'eu_sales', 1)
top_in_regions('genre', 'jp_sales', 2)

В Северной Америке самый популярный жанр за актуальный перио стал Shooter, на втором месте Action, самый не популярный это Misc. В Евпоре 2первых мета такиже как в Северной америке, только Action на первом месте, а Shooter на втором, самый же не популярный Rasing. В Японии, как всегда все наоборот, самый популярный Role-Playing, а Shooter на последнем месте из топ-5 жанров.

### Оценка влияния рейтинга ESRB на продажи в отдельном регионе

In [ ]:
fig, axes = plt.subplots(1, 3, figsize=(25,6))
top_in_regions('rating', 'na_sales', 0)
top_in_regions('rating', 'eu_sales', 1)
top_in_regions('rating', 'jp_sales', 2)

В Северной Америке и Европе опять похожие распределения: больше всего продаж с рейтингом М - для взрослых, меньше всего для E10+ -для всех от 10 лет и старше. В Японии больше всего продаж у игр без категории (скорее всего они просто не вводят эти категории, потому что рейтинг Американский) 

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Хорошая наблюдательность, действительно - ESRB является американской системой оценки.
    
</div>

Вывод:

- Типичный пользователь региона Северная Америка: выбирает shooter(стрелялку) с рейтингом "для взрослых" и играет на PS4 и XOne;
- Типичный пользователь региона Европа: выбирает экшн с рейтингом "для взрослых" и играет на PS4;
- Типичный пользователь региона Япония: выбирает ролевые игры с рейтингом без рейтинга и играет на 3DS.

<div class="alert alert-warning">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Совет**

Сами по себе выводы в работе хороши, но они построены на всем периоде :)
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Актуализировала. Конечно так попали реально более новые платформы и более популярные игры </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Успех**

Супер, теперь выводы можно использовать для прогноза
    
</div>

## Проверка гипотез

### Проверка гипотезы: средние пользовательские рейтинги платформ Xbox One и PC одинаковые

Сформулируем гипотезы:

Нулевая гипотеза H0 - средние пользовательские рейтинги платформ Xbox One и PC одинаковые. Альтернативная гипотеза Н1 - средние пользовательские рейтинги платформ Xbox One и PC неодинаковые.

Создадим 2 таблицы с выборкой по каждому платформе. Рейтинг 0 также учтем в срезе, поскольку нулевой рейтинг у игр реально может быть, а пропуски нулями мы не заполняли.

In [ ]:
xbox_one_platform = actual_period_data.query('platform == "XOne" & user_score >= 0')
pc_platform = actual_period_data.query('platform == "PC" & user_score >= 0')

Проверка нормальности распределения выборок

Построим распределения наших выборок и сравним с нормальным

In [ ]:
fig, ax = plt.subplots(1 ,2, figsize=(15,5))
sns.distplot(xbox_one_platform['user_score'], ax=ax[0])
sns.distplot(pc_platform['user_score'], ax=ax[1])
ax[0].set(xlabel='Оценка пользователей', title = 'Xbox One' )
ax[1].set(xlabel='Оценка пользователей', title = 'PC');

Для точного принятия решения, проведем еще одну проверку на "нормальность" распределения, на qq-графиках (квантиль-квантиль)

In [ ]:
plt.figure(figsize=(12,8))
plt.subplot(2,2,1)
st.probplot(xbox_one_platform['user_score'], dist = 'norm', plot=plt)
plt.subplot(2,2,2)
st.probplot(pc_platform['user_score'], dist = 'norm', plot=plt);

Красная линия на графиках - qq график нормального распределения, наши же распределения далеки от прямой линии.

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Успех**

Хорошая подготовка к проверке гипотезы, молодец
    
</div>

Поскольку распределения не являются нормальными будем использовать критерий Манна-Уитни для проверки гипотез. 
Критический уровень значимости задаем 0.05 (чтобы правильность вывода была не менее 95%)

In [ ]:
alpha = 0.05
results = st.mannwhitneyu(xbox_one_platform['user_score'], pc_platform['user_score'])
print('p-значение:', results.pvalue)
if (results.pvalue < alpha):
    print('Отвергаем нулевую гипотезу')
else:
    print('Принимаем нулевую гипотезу')

Принимаем нулевую гипотезу, т.е. средние пользовательские рейтинги платформ Xbox One и PC одинаковые.

In [ ]:
print('Средняя оценка пользователей игр Xbox One  - {:.2f}'.format(xbox_one_platform['user_score'].mean()))
print('Средняя оценка пользователей игр PC - {:.2f}'.format(pc_platform['user_score'].mean()))

И правда, средни оценки очень близки

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

У тебя исказились выводы, так как ты смотришь данные за все года :)
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Поправила, но результат остался такой же) </div>

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Ошибка**

Неа, ты взяла X360, а не XOne :)
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Поправила, и результат изменился. Полезно такое наблюдать) Спасибо, буду внимательнее </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 3</b> </font>
    
**Успех**

Отлично! 
    
</div>

### Проверка гипотезы: средние пользовательские рейтинги жанров Action и Sports разные.

Сформулируем гипотезы:

Нулевая гипотеза H0 - средние пользовательские рейтинги жанров Action и Sports одинаковые. Альтернативная гипотеза Н1 - средние пользовательские рейтинги жанров Action и Sports разные.

Создадим 2 таблицы с выборкой по каждому жанру. Рейтинг 0 также учтем в срезе, поскольку нулевой рейтинг у игр реально может быть, а пропуски нулями мы не заполняли.

In [ ]:
action_games = actual_period_data.query('genre == "Action" & user_score >= 0')
sports_games = actual_period_data.query('genre == "Sports" & user_score >= 0')

Проведем проверка нормальности распределения выборок, как в предыдущем пункте

In [ ]:
fig, ax = plt.subplots(1 ,2, figsize=(15,5))
sns.distplot(action_games['user_score'], ax=ax[0])
sns.distplot(sports_games['user_score'], ax=ax[1])
ax[0].set(xlabel='Оценка пользователей', title = 'Экшн игры' )
ax[1].set(xlabel='Оценка пользователей', title = 'Спортивные игры');

In [ ]:
plt.figure(figsize=(12,8))
plt.subplot(2,2,1)
st.probplot(action_games['user_score'], dist = 'norm', plot=plt)
plt.subplot(2,2,2)
st.probplot(sports_games['user_score'], dist = 'norm', plot=plt);

Поскольку распределения не являются нормальными(хотя жанр спорт очень близок к нормальному) будем использовать критерий Манна-Уитни для проверки гипотез (более ниверсальный). 
Критический уровень значимости задаем 0.05 (чтобы правильность вывода была не менее 95%)

In [ ]:
alpha = 0.05
results = st.mannwhitneyu(action_games['user_score'], sports_games['user_score'])
print('p-значение:', results.pvalue)
if (results.pvalue < alpha):
    print('Отвергаем нулевую гипотезу')
else:
    print('Принимаем нулевую гипотезу')

Отвергаем нулевую гипотезу, то есть рейтинги жанров экшн и спортивных игр разные

Посмотрим средние оценки пользователей по данным жанрам

In [ ]:
print('Средняя оценка пользователей игр жанра экшн  - {:.2f}'.format(action_games['user_score'].mean()))
print('Средняя оценка пользователей жанра спорт - {:.2f}'.format(sports_games['user_score'].mean()))

Действительно разница средних оценок игр жанров экшн и спортивных игр есть.

<div class="alert alert-danger">
<font size="5"<b>Комментарий ревьюера</b> </font>
    
**Ошибка**

Здесь аналогично - выводы исказились из-за использования данных за все года
    
</div>


<div class="alert alert-info"> <b>Комментарий студента:</b> Поправила, а тут вот резульат изменился </div>

<div class="alert alert-success">
<font size="5"<b>Комментарий ревьюера 2</b> </font>
    
**Успех**

Тут все верно :) 
    
</div>

## Общий вывод

 По анализу можно сделать вывод, что самое сильное влияние на итоговую выручку оказывают, в большинстве случаев, оценки критиков в большей степени чем оценки пользователей. 

Пользователи игр в Европе и Северной Америке в среднем схожи, по жанрам и категориям игр, пользователи же из Японии значительно отличаются. Надо определить политику на какой рынок нацелить продажи. Так типичный пользователь региона Северная Америка выбирает шутер с рейтингом "для взрослых" и играет на XOne или PS4; типичный пользователь региона Европа: выбирает экшн с рейтингом "для взрослых" и играет на PS4; типичный пользователь региона Япония: выбирает ролевые игры с рейтингом без рейтинга и играет на 3DS.

В целом рынок компьютерных игр для игровых платформ идет на убыль по продажам игр. Тут могут играть роль развитие мобильных игр, а также несанкционированное скачbвание игр с пиратских ресурсов.  С 2008 произошел резкий спад релиза игр. За весь период  приставок самые популярные оказались: PS2, X360, PS3, WII и DS, PS для японского рынка. Также выявили,что средняя продолжительность жизни игровой приставки составляет порядка 10 лет. Самый пик продаж наступает примерно через 3-5 лет после выпуска игры. Платформа XOne набирает полулярность среди жителей Севреной Америки и Европы.

Самый  за последние 10 лет это жанр Action. Растет в продажах жанр shooter. Однако производство игр в жанрах Action или Shooter обходится сильно дороже других игр.